admin-ui/plugins/fido/components/DynamicConfiguration.tsx

```diff
@@ -11,49 +11,61 @@ import GluuProperties from 'Routes/Apps/Gluu/GluuProperties'
 import GluuTypeAhead from 'Routes/Apps/Gluu/GluuTypeAhead'
 import { fidoConstants, validationSchema } from '../helper'
 import { transformToFormValues, getAvailableHintOptions, getEmptyDropdownMessage } from '../helper'
+import type {
+  DynamicConfigurationProps,
+  DynamicConfigFormValues,
+  DynamicConfigFormik,
+  DropdownOption,
+  FormData,
+} from './types/DynamicConfiguration'
 
-function DynamicConfiguration({ fidoConfiguration, handleSubmit }) {
+function DynamicConfiguration({
+  fidoConfiguration,
+  handleSubmit,
+}: DynamicConfigurationProps): JSX.Element {
   const { fido } = fidoConfiguration
 
-  const [modal, setModal] = useState(false)
+  const [modal, setModal] = useState<boolean>(false)
 
-  const toggle = useCallback(() => {
+  const toggle = useCallback((): void => {
     setModal((prev) => !prev)
   }, [])
 
-  const formik = useFormik({
-    initialValues: transformToFormValues(fido, fidoConstants.DYNAMIC),
+  const formik: DynamicConfigFormik = useFormik<DynamicConfigFormValues>({
+    initialValues: transformToFormValues(fido, fidoConstants.DYNAMIC) as DynamicConfigFormValues,
     onSubmit: toggle,
-    validationSchema: validationSchema[fidoConstants.VALIDATION_SCHEMAS.DYNAMIC_CONFIG],
+    validationSchema: validationSchema.dynamicConfigValidationSchema,
   })
 
-  const submitForm = useCallback(() => {
+  const submitForm = useCallback((): void => {
     toggle()
-    handleSubmit(formik.values)
-  }, [handleSubmit, toggle, formik])
+    handleSubmit(formik.values as unknown as FormData)
+  }, [handleSubmit, toggle, formik.values])
 
```

Comment:
Remove unsafe double-cast; cast directly to the aliased form type

The unknown → FormData cast is unnecessary and weakens type-safety.


```diff
-  const submitForm = useCallback((): void => {
+  const submitForm = useCallback((): void => {
     toggle()
-    handleSubmit(formik.values as unknown as FormData)
-  }, [handleSubmit, toggle, formik.values])
+    handleSubmit(formik.values as FidoFormData)
+  }, [handleSubmit, toggle, formik.values])
```